typescript/cli/src/context/altvm.ts

```diff
@@ -0,0 +1,210 @@
+import { password } from '@inquirer/prompts';
+
+import {
+  CosmosNativeProvider,
+  CosmosNativeSigner,
+} from '@hyperlane-xyz/cosmos-sdk';
+import {
+  AnyProtocolReceipt,
+  AnyProtocolTransaction,
+  ChainMap,
+  ChainMetadataManager,
+  ProtocolMap,
+  isJsonRpcSubmitterConfig,
+} from '@hyperlane-xyz/sdk';
+import { AltVM, type MINIMUM_GAS, ProtocolType } from '@hyperlane-xyz/utils';
+
+import { ExtendedChainSubmissionStrategy } from '../submitters/types.js';
+
+import { SignerKeyProtocolMap } from './types.js';
+
+// ### ALL Alt VM PROTOCOLS ARE REGISTERED HERE ###
+const ALT_VM_SUPPORTED_PROTOCOLS: ALT_VM_PROTOCOL = {
+  [ProtocolType.CosmosNative]: {
+    provider: CosmosNativeProvider,
+    signer: CosmosNativeSigner,
+    gas: {
+      CORE_DEPLOY_GAS: (1e6).toString(),
+      WARP_DEPLOY_GAS: (3e6).toString(),
+      TEST_SEND_GAS: (3e5).toString(),
+      AVS_GAS: (3e6).toString(),
+    },
+  },
+  // [NEW PROTOCOL]: {...}
+};
+
+type ALT_VM_PROTOCOL = ProtocolMap<{
+  provider: AltVM.IProviderConnect;
+  signer: AltVM.ISignerConnect<AnyProtocolTransaction, AnyProtocolReceipt>;
+  gas: MINIMUM_GAS;
+}>;
+
+class AltVMFactory implements AltVM.IAltVMFactory {
+  public getSupportedProtocols(): ProtocolType[] {
+    return Object.keys(ALT_VM_SUPPORTED_PROTOCOLS) as ProtocolType[];
+  }
+
+  public supports(protocol: ProtocolType) {
+    return !!ALT_VM_SUPPORTED_PROTOCOLS[protocol];
+  }
+
+  public getMinGas(protocol: ProtocolType) {
+    const protocolDefinition = ALT_VM_SUPPORTED_PROTOCOLS[protocol];
+
+    if (!protocolDefinition) {
+      throw new Error(`Protocol type ${protocol} not supported in AltVM`);
+    }
+
+    return protocolDefinition.gas;
+  }
+}
```

Comment:
Isn't this class a bit misleading? it is called `AltVMFactory` but looking at it as a whole it just looks like a wrapper on the `ALT_VM_SUPPORTED_PROTOCOLS` to get gas and supported protocols data but nothing is done about the signers and providers in here, wouldn't it be better if this class also allowed to get providers and signers so on the cli side you can use only this class instead of 3 different classes depending what you want to achieve?

Reply:
To be clear I'm not saying that you should eliminate the `AltVMProviderFactory` and `AltVMSignerFactory` classes but instead those 2 classes should be inside the `AltVMFactory` so you can get a signer or provider